managers/asset-operations-manager.js

```diff
@@ -30,6 +31,68 @@ class AssetOperationsManager {
         this.inputService = services.inputService;
     }
 
+    /**
+     * Checks if given UAL is valid.
+     * @async
+     * @param {string} UAL - Universal Asset Locator.
+     * @param {Object} [options={}] - Additional options - currently only blockchain option expected.
+     * @returns {boolean} UAL have passed validation.
+     * @throws {Error} Throws an error if UAL validation fails.
+     * @example did:dkg:otp:2043/0x5cac41237127f94c2d21dae0b14bfefa99880630/1985318
+     */
+    async isValidUAL(UAL, options = {}) {
+        if (typeof UAL !== 'string' || UAL.trim() === '') {
+            throw Error('UAL must be a non-empty string.');
+        }
+
+        const blockchain = this.inputService.getBlockchain(options);
+        this.validationService.validateIsValidUAL(blockchain);
+
+        const parts = UAL.split('/');
+        if (parts.length !== 3) {
+            throw Error('UAL format is incorrect.');
+        }
+
+        const prefixes = parts[0].split(':');
+        if (prefixes.length !== 3 && prefixes.length !== 4) {
+            throw Error('Prefix format in UAL is incorrect.');
+        }
+
+        if (prefixes[0] !== 'did') {
+            throw Error('Invalid DID prefix.');
+        }
+
+        if (prefixes[1] !== 'dkg') {
+            throw Error('Invalid DKG prefix.');
+        }
+
+        if (prefixes[2] !== blockchain.name) {
```

Comment:
If we use blockchain name with id, and old ual would this cause it to fail?

Reply:
Chain ID is separate, like: did:dkg:otp:2043

Reply:
Yeah but on node its part of blockchain name, I assumed we will also do it here like that so blockchainame will be otp:2043